### PR TITLE
Enable `disallow_untyped_calls` for mypy

### PR DESCRIPTION
**Pull request by 29Takuya**
## Motivation

https://github.com/optuna/optuna/issues/3579

## Description of the changes

Enable `disallow_untyped_calls` option for mypy.

### CODE REVIEW

**Comment by contramundum53** [comment]
@not522 Could you review this PR?

File: tests/visualization_tests/matplotlib_tests/test_pareto_front.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def _check_data(figure: Axes, axis: str, expected: Sequence[int]) -> None:
             )
         )
     )
-    numpy.ma.testutils.assert_array_equal(actual, expected)
+    numpy.ma.testutils.assert_array_equal(actual, expected)  # type: ignore[no-untyped-call]
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_pareto_front.py:52]: I suppose we don't need [`numpy.ma`](https://numpy.org/doc/stable/reference/maskedarray.generic.html) module, which is for masked array.

How about replacing the line with 
```suggestion
    numpy.testing.assert_array_equal(actual, expected)
```

Comment (29Takuya) [reply tests/visualization_tests/matplotlib_tests/test_pareto_front.py:52]: Thank you for your suggestion!
I confirmed that addresses the issue.

**Review by nzw0301 — APPROVED** [review]
Brilliant. Thanks!

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**